lib/fog/azurerm/models/storage/file.rb

```
@@ -4,147 +4,386 @@ class AzureRM
-          hash['metadata'] = blob.metadata
-          return hash unless blob.respond_to?('properties')
-
-          hash['last_modified'] = blob.properties[:last_modified]
-          hash['etag'] = blob.properties[:etag]
-          hash['lease_duration'] = blob.properties[:lease_duration]
-          hash['lease_status'] = blob.properties[:lease_status]
-          hash['lease_state'] = blob.properties[:lease_state]
-          hash['content_length'] = blob.properties[:content_length]
-          hash['content_type'] = blob.properties[:content_type]
-          hash['content_encoding'] = blob.properties[:content_encoding]
-          hash['content_language'] = blob.properties[:content_language]
-          hash['content_disposition'] = blob.properties[:content_disposition]
-          hash['content_md5'] = blob.properties[:content_md5]
-          hash['cache_control'] = blob.properties[:cache_control]
-          hash['sequence_number'] = blob.properties[:sequence_number]
-          hash['blob_type'] = blob.properties[:blob_type]
-          hash['copy_id'] = blob.properties[:copy_id]
-          hash['copy_status'] = blob.properties[:copy_status]
-          hash['copy_source'] = blob.properties[:copy_source]
-          hash['copy_progress'] = blob.properties[:copy_progress]
-          hash['copy_completion_time'] = blob.properties[:copy_completion_time]
-          hash['copy_status_description'] = blob.properties[:copy_status_description]
-          hash['accept_ranges'] = blob.properties[:accept_ranges]
-          hash
+        # Upload a page blob
+        def save_page_blob(blob_size, options)
+          # Initiate the upload
+          service.create_page_blob(directory.key, key, blob_size, options)
+
+          # Uploading content
+          # TODO: Upload content in parallel using threads
+          if body.respond_to?(:read)
+            if body.respond_to?(:rewind)
+              begin
+                body.rewind
+              rescue
+                nil
```

The exception is ignored by the code , which is not a good practice. 

If we are rescuing the exception then we should at least show the message here.